core/trino-spi/src/main/java/io/trino/spi/block/RunLengthEncodedBlock.java

```diff
@@ -64,6 +65,7 @@ public RunLengthEncodedBlock(Block value, int positionCount)
         }
 
         this.positionCount = positionCount;
+        this.isNull = positionCount > 0 && value.isNull(0);
```

Comment:
undo: this will cause lazy block to be loaded.
Maybe do something similar as with `io.trino.spi.block.DictionaryBlock#DictionaryBlock(int, int, io.trino.spi.block.Block, int[], boolean, boolean, io.trino.spi.block.DictionaryId)`?

Reply:
Do we have a case where we create RunLengthEncodedBlock out of a LazyBlock ? So far I didn't see one.
We already asserted the block to contain single position. I'm not sure why we would want to create a LazyBlock out of a single valued Block.

Reply:
> Do we have a case where we create RunLengthEncodedBlock out of a LazyBlock ? So far I didn't see one.

It breaks contract (e.g. forces loading in constructor). You could imagine joining single position with multiple build rows.